third_party/amd/lib/TritonAMDGPUToLLVM/TargetInfo.cpp

```diff
@@ -200,15 +200,19 @@ bool TargetInfo::warpReduce(RewriterBase &rewriter, Location loc,
                             unsigned numLaneToReduce,
                             unsigned interleave) const {
   auto b = TritonLLVMOpBuilder(loc, rewriter);
-  if (numLaneToReduce != 64)
-    return false;
 
-  if (!llvm::is_contained(
-          {ISAFamily::CDNA2, ISAFamily::CDNA3, ISAFamily::CDNA4},
-          getISAFamily())) {
+  if (!(numLaneToReduce == 64 &&
+            llvm::is_contained(
+                {ISAFamily::CDNA2, ISAFamily::CDNA3, ISAFamily::CDNA4},
+                getISAFamily()) ||
+        numLaneToReduce == 32 &&
+            llvm::is_contained({ISAFamily::RDNA3}, getISAFamily()))) {
```

Comment:
Are we missing some parenthesis on both side of `||` here? I suppose it's like `if (!((64 && cdna) || (32 && rdna)))`?

Reply:
Should be fine given the usual operator precedence rules I think? But yes this conditional is kind of a mess...